packages/block-editor/src/components/button-block-appender/index.js

```diff
@@ -52,30 +59,35 @@ function ButtonBlockAppender(
 					);
 				}
 				const isToggleButton = ! hasSingleBlockType;
-				return (
-					<Tooltip text={ label }>
-						<Button
-							ref={ ref }
-							onFocus={ onFocus }
-							tabIndex={ tabIndex }
-							className={ classnames(
-								className,
-								'block-editor-button-block-appender'
-							) }
-							onClick={ onToggle }
-							aria-haspopup={
-								isToggleButton ? 'true' : undefined
-							}
-							aria-expanded={
-								isToggleButton ? isOpen : undefined
-							}
-							disabled={ disabled }
-							label={ label }
-						>
+
+				const inserterButton = (
+					<Button
+						ref={ ref }
+						onFocus={ onFocus }
+						tabIndex={ tabIndex }
+						className={ classnames(
+							className,
+							'block-editor-button-block-appender'
+						) }
+						onClick={ onToggle }
+						aria-haspopup={ isToggleButton ? 'true' : undefined }
+						aria-expanded={ isToggleButton ? isOpen : undefined }
+						disabled={ disabled }
+						label={ label }
+					>
+						{ ! hasSingleBlockType && (
 							<VisuallyHidden as="span">{ label }</VisuallyHidden>
-							<Icon icon={ create } />
-						</Button>
-					</Tooltip>
+						) }
+						<Icon icon={ create } />
+						{ hasSingleBlockType && <span>{ label } </span> }
+					</Button>
+				);
+				return (
+					<AddTooltipForIconButton
```

Comment:
Does it have to be a separate component? Imperative component name feels strange. `inserterButton` could be a variable instead of a constant and then we could just do `if ( hasTooltip )` in this function.